jac-uk/digital-platform
# #757 Optimise sentry configuration

## What's included?
1. Add sentry SDK (v7.12.1) to google cloud functions.
    - Add `wrapFunction` to callable functions.
    - Add `Sentry.GCPFunction.wrapEventFunction` to background functions.

2. Set sentry dsn to config by running the command: `firebase functions:config:set sentry.dsn=<sentry_dsn>`.

## Who should test?
✅ Developers

## How to test?
1. Use Postman to test the endpoints of callable functions.
2. Check if the errors are reported to Sentry.

## Risk - how likely is this to impact other areas?
🟢 No risk - this is a self-contained piece of work

## Additional context
Demo:

https://user-images.githubusercontent.com/79906532/190193712-3d8665a5-9484-454b-ae49-5ff761b9b2c5.mov

## Related permissions
No permission changes required

---
PREVIEW:DEVELOP
